### PR TITLE
Fix wso2/product-is#3459: Management console UI support for application side consent management

### PR DESCRIPTION
**Pull request by omindu**

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp [hunk 1]
```diff
@@ -1390,7 +1489,146 @@
...
+                        </table>
+                        <table style="padding-top: 15px; padding-bottom: 10px;" class="carbonFormTable" id="general_purposes_combo">
+                                <tr>
+                                    <td class="leftCol-med labelField"><fmt:message key="general.consent.purposes"/></td>
+                                </tr>
+                                <tr>
+                                    <td class="leftCol-med">
+                                            <select class="leftCol-med" id="general_purposes" name="general_purposes"
+                                            style=" margin-left: 5px; ">
+                                                <option value="not_selected">---Select---</option>
```
Comment (thusithak) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp]: Can't we add i18n here?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp [hunk 2]
```diff
@@ -1390,7 +1489,146 @@
...
+                                </script>
+                                <%}%>
+                                <%for (ApplicationPurpose generalPurpose : appGeneralPurposes) {%>
+                                <tr id="row_general_purpose_id_<%=generalPurpose.getId()%>">
+                                    <td><%=Encode.forHtmlContent(generalPurpose.getName())%></td>
+                                    <td><%=Encode.forHtmlContent(generalPurpose.getDescription())%></td>
+                                    <td><input style="width:30px" name="display_order_gen_purpose_id_<%=generalPurpose.getId()%>" 
+                                            type="number" min="0" type="text" value="<%=generalPurpose.getDisplayOrder()%>" autofocus=""></td>
+                                    <td><a class="icon-link" style="background-image: url(../admin/images/delete.gif)" 
+                                        onclick="removeGeneralPurposeRow('row_general_purpose_id_<%=generalPurpose.getId()%>')">Delete</a></td>
```
Comment (thusithak) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp]: Can't we add i18n here?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp [hunk 3]
```diff
@@ -1390,7 +1489,146 @@
...
+                                    <th class="leftCol-big"><fmt:message key="purpose"/></th>
+                                    <th class="leftCol-big"><fmt:message key="description"/></th>
+                                    <th class="leftCol-mid"><fmt:message key="display.order"/></th>                                    
+                                    <th class="leftCol-mid"><fmt:message key="remove"/></th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                                <%if (appGeneralPurposes == null || appGeneralPurposes.isEmpty()) {%>
+                                <script>
+                                    $(jQuery('#general_purposes_tbl')).hide();
```
Comment (thusithak) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/resources/web/application/configure-service-provider.jsp]: Use either 
`$('#general_purposes_tbl').hide();`
or
`jQuery('#general_purposes_tbl').hide();`

Both methods achieve the same, no need to wrap a jQuery object again with jQuery

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 4]
```diff
@@ -1438,15 +1446,85 @@ public void update(HttpServletRequest request) {
                 alwaysSendMappedLocalSubjectId != null
                 && "on".equals(alwaysSendMappedLocalSubjectId) ? true : false);
 
-        // TODO: Update the configs from request once the SP consent purpose UI is implemented.
+        boolean consentEnabled = request.getParameter("is_consent_enabled") != null;
```
Comment (nuwandiw) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java]: Better to define a constant for "is_consent_enabled"

**Pull request merged**